app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    salt = Pharos::Salt.new
+    available_roles = %i(master)
+    salt.minions.keys.each do |minion|
+      Minion.transaction do
+        Minion.find_or_create_by!(hostname: minion).assign_role(available_roles)
```

Comment:
I think there is a bug here. I think ```assign_roles``` code assumes that any assigned role is removed from the available_roles array, which is true while inside the method but since Ruby is pass-by-value, that change isn't reflected on the available_roles here. So all the roles are present in every iteration which might result in multiple masters.